libnavui-voice/src/main/java/com/mapbox/navigation/ui/voice/api/MapboxOnboardSpeechPlayer.kt

```diff
@@ -0,0 +1,134 @@
+package com.mapbox.navigation.ui.voice.api
+
+import android.content.Context
+import android.os.Bundle
+import android.speech.tts.TextToSpeech
+import android.speech.tts.UtteranceProgressListener
+import com.mapbox.navigation.ui.base.api.voice.SpeechPlayer
+import com.mapbox.navigation.ui.base.api.voice.SpeechPlayerCallback
+import com.mapbox.navigation.ui.base.model.voice.Announcement
+import com.mapbox.navigation.ui.base.model.voice.SpeechState
+import java.util.Locale
+
+/**
+ * Offline implementation of [SpeechPlayer].
+ * @property context Context
+ * @property language [Locale] language (ISO 639)
+ */
+internal class MapboxOnboardSpeechPlayer(
+    private val context: Context,
+    private val language: String
+) : SpeechPlayer {
+
+    private var isLanguageSupported: Boolean = false
+    private val textToSpeech = TextToSpeech(context.applicationContext) { status ->
+        if (status == TextToSpeech.SUCCESS) {
+            initializeWithLanguage(Locale(language))
+        }
+    }
+    private var volumeLevel: Float = DEFAULT_VOLUME_LEVEL
+    private var clientCallback: SpeechPlayerCallback? = null
+    private var currentPlay: SpeechState.ReadyToPlay? = null
+
+    /**
+     * Given [SpeechState.ReadyToPlay] [Announcement] the method will play the voice instruction.
+     * If a voice instruction is already playing or other announcement are already queued,
+     * the given voice instruction will be queued to play after.
+     * @param state SpeechState Play Announcement object including the announcement text
+     * and optionally a synthesized speech mp3.
+     * @param callback
+     */
+    override fun play(state: SpeechState.ReadyToPlay, callback: SpeechPlayerCallback) {
+        clientCallback = callback
+        check(currentPlay == null) {
+            "Only one announcement can be played at a time."
```

Comment:
The queue prevents this from happening in the hybrid player, right?

Reply:
Correct. That's addressing @LukasPaczos's comment https://github.com/mapbox/mapbox-navigation-android/pull/3977#discussion_r574656472 

Since`Onboard` and `Offboard` are internal implementations, we can consider that if `play` is invoked while another instruction is playing a bug, as it's controlled by us in the `Hybrid`.